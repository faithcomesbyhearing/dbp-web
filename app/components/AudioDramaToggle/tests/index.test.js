import React from 'react';
import '@testing-library/jest-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import { AudioDramaToggle } from '../index';

const props = {
	audioType: 'audio_drama',
	availableAudioTypes: ['audio_drama', 'audio'],
	dispatch: jest.fn(),
};

describe('<AudioDramaToggle />', () => {
	it('Expect to match snapshot for drama audio', () => {
		const { asFragment } = render(<AudioDramaToggle {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Expect to match snapshot for plain audio', () => {
		const { asFragment } = render(<AudioDramaToggle {...props} audioType="audio" />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Expect to match snapshot for only plain audio available', () => {
		const { asFragment } = render(
		<AudioDramaToggle {...props} availableAudioTypes={['audio']} />,
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('Expect to match snapshot for only drama audio available', () => {
		const { asFragment } = render(
		<AudioDramaToggle {...props} availableAudioTypes={['audio_drama']} />,
		);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render two buttons and handle clicks correctly', () => {
		render(<AudioDramaToggle {...props} />);

		const dramaButton = screen.getByText('Drama');
		const nonDramaButton = screen.getByText('Non-Drama');

		expect(dramaButton).toBeInTheDocument();
		expect(nonDramaButton).toBeInTheDocument();

		// Simulate clicks and check dispatch calls
		fireEvent.click(dramaButton);
		expect(props.dispatch).toHaveBeenCalledTimes(0); // drama is already selected

		fireEvent.click(nonDramaButton);
		expect(props.dispatch).toHaveBeenCalledTimes(1); // switching to non-drama

		fireEvent.click(nonDramaButton);
		expect(props.dispatch).toHaveBeenCalledTimes(2); // non-drama
	});
});
