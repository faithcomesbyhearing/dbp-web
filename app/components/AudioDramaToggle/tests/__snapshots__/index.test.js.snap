// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AudioDramaToggle /> Expect to match snapshot for drama audio 1`] = `
<DocumentFragment>
  <div
    class="audio-drama-toggle-container"
  >
    <button
      class="audio-drama-toggle-button active"
      id="drama-button"
      title="Drama"
      type="button"
      value="audio_drama"
    >
      Drama
    </button>
    <button
      class="audio-drama-toggle-button"
      id="non-drama-button"
      title="Non-Drama"
      type="button"
      value="audio"
    >
      Non-Drama
    </button>
  </div>
</DocumentFragment>
`;

exports[`<AudioDramaToggle /> Expect to match snapshot for only drama audio available 1`] = `
<DocumentFragment>
  <div
    class="audio-drama-toggle-container"
  >
    <button
      class="audio-drama-toggle-button active"
      id="drama-button"
      title="Drama"
      type="button"
      value="audio_drama"
    >
      Drama
    </button>
    <button
      class="audio-drama-toggle-button disabled"
      id="non-drama-button"
      title="Non-Drama"
      type="button"
      value="audio"
    >
      Non-Drama
    </button>
  </div>
</DocumentFragment>
`;

exports[`<AudioDramaToggle /> Expect to match snapshot for only plain audio available 1`] = `
<DocumentFragment>
  <div
    class="audio-drama-toggle-container"
  >
    <button
      class="audio-drama-toggle-button active disabled"
      id="drama-button"
      title="Drama"
      type="button"
      value="audio_drama"
    >
      Drama
    </button>
    <button
      class="audio-drama-toggle-button"
      id="non-drama-button"
      title="Non-Drama"
      type="button"
      value="audio"
    >
      Non-Drama
    </button>
  </div>
</DocumentFragment>
`;

exports[`<AudioDramaToggle /> Expect to match snapshot for plain audio 1`] = `
<DocumentFragment>
  <div
    class="audio-drama-toggle-container"
  >
    <button
      class="audio-drama-toggle-button"
      id="drama-button"
      title="Drama"
      type="button"
      value="audio_drama"
    >
      Drama
    </button>
    <button
      class="audio-drama-toggle-button active"
      id="non-drama-button"
      title="Non-Drama"
      type="button"
      value="audio"
    >
      Non-Drama
    </button>
  </div>
</DocumentFragment>
`;
